src/System.IO.Compression/src/System/IO/Compression/DeflateManaged/InflaterManaged.cs

```diff
@@ -134,10 +137,26 @@ public int Inflate(byte[] bytes, int offset, int length)
                     _formatReader.Validate();
                 }
             }
+            count = RestrictStreamWithUnCompressedSize(count, out bool wasOverLimit);
+
+            if (wasOverLimit)
+                throw new InvalidDataException(SR.GenericInvalidData);
```

Comment:
You may want to prevent further calls from doing any work once you've reached the limit.  Ideally you implement the limit such that you'll never read / decompress more than that into the callers byte array.

Reply:
That is one main reasons of throwing here, this method called only for copying one chunk of decompressed output  of inflator into returning stream, at this point there is no other way to stop decompressing other than throwing. (maybe better to say I didn't found any other way except throwing)

Reply:
> That is one main reason of throwing here, this method called only for copying one chunk of decompressed output of inflator into returning stream, at this point there is no other way to stop decompressing other than throwing

The above comment based on the previous solution when i was restricting bytes on the deflate stream, but for this solution when restricting bytes at inflator level there is flag bytes indicating/used to stop inflation. So now we are not throwing just resticting uncompressed stream as you suggested.